compiler/rustc_lint/src/internal.rs

```diff
@@ -267,3 +267,42 @@ impl EarlyLintPass for LintPassImpl {
         }
     }
 }
+
+declare_tool_lint! {
+    pub rustc::EXISTING_DOC_KEYWORD,
+    Allow,
+    "Check that documented keywords in std and core actually exist",
+    report_in_external_macro: true
+}
+
+declare_lint_pass!(ExistingDocKeyword => [EXISTING_DOC_KEYWORD]);
+
+fn is_doc_keyword(s: Symbol) -> bool {
+    s <= kw::Union
+}
+
+impl EarlyLintPass for ExistingDocKeyword {
+    fn check_attribute(&mut self, cx: &EarlyContext<'_>, attr: &rustc_ast::Attribute) {
+        if attr.has_name(sym::keyword) {
+            if let Some(v) = attr.value_str() {
+                // if v.to_string() == "lol" {
+                //     panic!("invalid!");
+                // }
```

Comment:
```suggestion
```

Reply:
This is a draft, I want this PR to make CI fail to be sure that it is doing what's supposed to do.

Reply:
oh, didn't realize :+1: